src/recon.rs

```diff
@@ -3441,29 +3469,37 @@ pub(crate) unsafe fn rav1d_recon_b_inter<BD: BitDepth>(
                 .interintra
                 .as_mut_ptr();
             let mut angle = 0;
-            let mut top_sb_edge: *const BD::Pixel = 0 as *const BD::Pixel;
-            if t.by & (*f).sb_step - 1 == 0 {
-                top_sb_edge = (*f).ipred_edge[0] as *mut BD::Pixel;
+            let top_sb_edge_slice = if t.by & (*f).sb_step - 1 == 0 {
+                let mut top_sb_edge: *const BD::Pixel = (*f).ipred_edge[0] as *const BD::Pixel;
                 let sby = t.by >> (*f).sb_shift;
                 top_sb_edge = top_sb_edge.offset(((*f).sb128w * 128 * (sby - 1)) as isize);
-            }
+                Some(slice::from_raw_parts(
+                    top_sb_edge,
+                    (*f).sb128w as usize * 128,
+                ))
+            } else {
+                None
+            };
             m = rav1d_prepare_intra_edges(
                 t.bx,
-                (t.bx > (*ts).tiling.col_start) as c_int,
+                t.bx > (*ts).tiling.col_start,
                 t.by,
-                (t.by > (*ts).tiling.row_start) as c_int,
+                t.by > (*ts).tiling.row_start,
                 (*ts).tiling.col_end,
                 (*ts).tiling.row_end,
                 0 as EdgeFlags,
                 dst,
                 (*f).cur.stride[0],
-                top_sb_edge,
+                top_sb_edge_slice,
                 m,
                 &mut angle,
                 bw4,
                 bh4,
                 0 as c_int,
-                tl_edge,
+                &mut BD::select_mut(&mut t.scratch.c2rust_unnamed_0.interintra_edge)
+                    .0
+                    .edge,
+                32,
```

Comment:
I think it would be better to, where `tl_edge` is defined, rewrite that section as:

```rust
let tl_edge_array = BD::select_mut(&mut t.scratch.c2rust_unnamed_0.interintra_edge).0.edge;
let tl_edge_offset = 32;
let tl_edge = tl_edge_array[tl_edge_offset..].as_mut_ptr();
```

and then make the call here with `tl_edge_array, tl_edge_offset`.

Otherwise, it's hard to realize they're using the same offset and starting from the same array, and eventually we'll also need to fix the negative indices in other uses of `tl_edge`, so `tl_edge_offset` will be useful there.

And this same comment also applies to `edge`, the other version of this.

Reply:
I made modifications to avoid duplication of offset constants and array/ptr derivation. However, I have one question regarding your suggestion: does `let tl_edge_array = BD::select_mut(&mut t.scratch.c2rust_unnamed_0.interintra_edge).0.edge;` result in the array `edge` being copied into a new array `tl_edge_array`?

Reply:
No, as you can see from `unsafe fn select_mut<T>(bd: &mut BitDepthUnion<T>) -> &mut T::T<Self> where ...`'s signature, it returns a `&mut`, so the array isn't being copied.